examples/hybrid_jobs/6_QNSPSA_optimizer_with_embedded_simulator/qnspsa_with_embedded_simulator.ipynb

```diff
@@ -0,0 +1,1360 @@
+    "    \n",
+    "# Defining ansatz and loss function\n",
+    "@qml.qnode(dev)\n",
+    "def cost_function(params):\n",
+    "    qaoa_circuit(params, n_qubits, depth)\n",
+    "    return qml.expval(cost_h) \n",
+    "\n",
+    "# Initializing parameters\n",
+    "params_init = 2 * np.pi * (np.random.rand(2, depth) - 0.5) \n",
+    "# Visualizing the QAOA circuit \n",
+    "drawer = qml.draw(cost_function)\n",
+    "print(drawer(params_init))"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "1d2ff62d",
+   "metadata": {},
+   "source": [
+    "Now I would like to see if my optimizer indeed reduces the loss."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 6,
+   "id": "54a85c9b",
+   "metadata": {},
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Step 0: cost = -2.5116\n",
+      "Step 40: cost = -2.7121\n",
+      "Step 80: cost = -2.7380\n",
+      "Step 120: cost = -2.9003\n",
+      "Step 160: cost = -2.9015\n",
+      "Step 200: cost = -2.9021\n",
+      "Step 240: cost = -2.9035\n",
+      "Step 280: cost = -2.9035\n"
+     ]
+    }
+   ],
+   "source": [
+    "# import the QN-SPSA optimizer\n",
+    "from source_scripts.QNSPSA import QNSPSA\n",
+    "# Optimize with QN-SPSA\n",
+    "opt = QNSPSA(lr=5e-2)\n",
+    "params = params_init\n",
+    "for i in range(300):\n",
+    "    params, loss = opt.step_and_cost(cost_function, params)\n",
+    "    if i % 40 == 0:\n",
+    "        print(f\"Step {i}: cost = {loss:.4f}\")"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "79788f27",
+   "metadata": {},
+   "source": [
+    "The analytical solution to the toy example gives a global minimum of -3. As the loss drops and finally converges close to the global minimum, we can tell that the optimizer works! "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "3dd69e15",
+   "metadata": {},
+   "source": [
+    "## Benchmarking\n",
+    "I will now proceed to benchmark the QN-SPSA optimizer's performance with the QAOA problem. I follow a similar procedure with Ref [1] to compare the performance of QN-SPSA, SPSA, vanilla gradient descent and quantum natural gradient descent, as they are similar to each other. "
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "b99980d2",
+   "metadata": {},
+   "source": [
+    "I choose to perform the benchmarking tests with Braket Jobs using the embedded simulator. When compared to the Braket notebooks, Braket Jobs has a few advantages for this benchmarking stage:\n",
+    "\n",
+    "- First, I would like my benchmarking tests to be reproducible, which requires a systematic way to track the logs and the outputs. This is easily done with Braket Jobs, but with the notebook, I would run into connection/permission timeout issues when the execution time of my program goes up to a few hours. I could easily lose the returned results or printed logs due to this issue. \n",
```

Comment:
"This is easily done with Braket Jobs, but with the notebook ... " : I'd break this into two sentences to not shed undeserved negative light on Braket Jobs in general. There is no reason to connect with "but".